### PR TITLE
filebrowser: 2.23.0 -> 2.31.0

### PR DESCRIPTION
**Pull request by HritwikSinghal**
## Description of changes

Updated the filebrowser from 2.23.0 to 2.31.0.

https://github.com/filebrowser/filebrowser/blob/master/CHANGELOG.md

## Things done

- updated the hashes
- updated version

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/filebrowser/default.nix
Comment (iblech) [inline pkgs/applications/networking/filebrowser/default.nix:58]: Welcome to the world of nixpkgs pull requests, and great that you want to maintain this package! You need to add yourself to `maintainers/maintainer-list.nix`, else your handle will not resolve :-)

What about the previous maintainer? Do you know that they no longer want to be listed? If not, I'd suggest to just add yourself (in alphabetic order) to them.

I'm not familiar with filebrowser, but this commit looks solid. Thank you for your work!

Comment (HritwikSinghal) [reply pkgs/applications/networking/filebrowser/default.nix]: Hi, i have added myself in nix-maintainer list and have also made the required changes. let me know if any more changes are needed. :)

**Comment by HritwikSinghal** [comment]
Hi @iblech , can we merge this PR. seems like @nielsegberts is no longer maintaining this package.

**Comment by HritwikSinghal** [comment]
@iblech can we merge this ?

File: pkgs/applications/networking/filebrowser/default.nix
Comment (Aleksanaa) [inline pkgs/applications/networking/filebrowser/default.nix:58]: Are you formatting this file?

```suggestion
    maintainers = with maintainers; [ nielsegberts oakenshield ];
```
```suggestion
    maintainers = with maintainers; [
      nielsegberts
      oakenshield
   ];
```

**Comment by HritwikSinghal** [comment]
@Aleksanaa formatted the code.

File: pkgs/applications/networking/filebrowser/default.nix
Comment (HritwikSinghal) [reply pkgs/applications/networking/filebrowser/default.nix]: i did not format it before. did it after this suggestion.

**Comment by Aleksanaa** [comment]
Please remove that `rec` as suggested by CI

**Comment by HritwikSinghal** [comment]
> Please remove that `rec` as suggested by CI

done :+1:

**Comment by Aleksanaa** [comment]
Please squash into two commits:
```
maintainers: add oakenshield
filebrowser: 2.23.0 -> 2.30.0
```

**Comment by HritwikSinghal** [comment]
> Please squash into two commits:
> 
> ```
> maintainers: add oakenshield
> filebrowser: 2.23.0 -> 2.30.0
> ```

Hi, did the squash and rebased on master.

**Comment by HritwikSinghal** [comment]
@Aleksanaa @iblech  can we merge this PR? are there any other check which need to be done?

**Comment by HritwikSinghal** [comment]
@Aleksanaa  @iblech   can we merge this PR? are there any other check which need to be done?

**Comment by HritwikSinghal** [comment]
image

**Comment by phanirithvij** [comment]
other prs touching filebrowser https://github.com/NixOS/nixpkgs/pull/289750 https://github.com/NixOS/nixpkgs/pull/306615

**Comment by phanirithvij** [comment]
And 2.31.0 https://github.com/filebrowser/filebrowser/releases/tag/v2.31.0

**Comment by HritwikSinghal** [comment]
> And 2.31.0 https://github.com/filebrowser/filebrowser/releases/tag/v2.31.0

needs go 1.23 which is not there in nixpkgs unstable yet. Will update filebrowser in new PR

**Comment by HritwikSinghal** [comment]
> other prs touching filebrowser #289750 #306615

Thanks, will take help of these PR's to create a module once this PR gets merged.

**Comment by HritwikSinghal** [comment]
updated to v2.31.0 by using `buildGo123Module` instead of `buildGoModule` as this version requires go 1.23.

File: pkgs/applications/networking/filebrowser/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/filebrowser/default.nix:19]:
```suggestion
  };

  frontend = buildNpmPackage rec {
```

File: pkgs/applications/networking/filebrowser/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/filebrowser/default.nix:63]: One new line to many

Comment (HritwikSinghal) [reply pkgs/applications/networking/filebrowser/default.nix]: removed it :+1:

File: pkgs/applications/networking/filebrowser/default.nix
Comment (HritwikSinghal) [reply pkgs/applications/networking/filebrowser/default.nix]: fixed it :+1:

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by phanirithvij** [comment]
> Thanks, will take help of these PR's to create a module once this PR gets merged.

There's already 4 people interested in implementing it https://github.com/NixOS/nixpkgs/issues/236165 I would suggest let's wait and see, and focus on bringing one of these 2 prs forward.
I'm just saying no need for more duplicated effort.

**Comment by HritwikSinghal** [comment]
Do i need to open another PR for backport? After cherry-pick, what should i do?

**Comment by SuperSandro2000** [comment]
Why do we want to backport it at all? it seems like a bigger update.

**Comment by HritwikSinghal** [comment]
> Why do we want to backport it at all? it seems like a bigger update.

ok, i thought since the backport tags were there we need to backport it. And NixOS 24.11 is due is 3 months anyways.

**Comment by HritwikSinghal** [comment]
> Why do we want to backport it at all? it seems like a bigger update.

can you also review this PR. https://github.com/NixOS/nixpkgs/pull/338990

I just ran the update script in the folder itself and it updated the version. This is a security issue release so can you quickly review and merge it.

**Comment by SuperSandro2000** [comment]
> This is a security issue release so can you quickly review and merge it.

please label them accordingly

**Pull request merged**
